python/ray/autoscaler/_private/commands.py

```diff
@@ -364,17 +364,21 @@ def remaining_nodes():
 
     def run_docker_stop(node, container_name):
         try:
-            exec_cluster(
-                config_file,
-                cmd=f"docker stop {container_name}",
-                run_env="host",
-                screen=False,
-                tmux=False,
-                stop=False,
-                start=False,
-                override_cluster_name=override_cluster_name,
-                port_forward=None,
-                with_output=False)
+            updater = NodeUpdaterThread(
+                node_id=node,
+                provider_config=config["provider"],
+                provider=provider,
+                auth_config=config["auth"],
+                cluster_name=config["cluster_name"],
+                file_mounts=config["file_mounts"],
+                initialization_commands=[],
+                setup_commands=[],
+                ray_start_commands=[],
+                runtime_hash="",
+                file_mounts_contents_hash="",
+                is_head_node=False,
+                docker_config=config.get("docker"))
+            _exec(updater, cmd=f"docker stop {container_name}", run_env="host")
```

Comment:
hmmm what does this change do?

Reply:
Before we ran `docker stop {container_name}` on the head node _every time_. We need to get an Updater _per worker_ and then run `docker stop {container_name}` on each node!